pkg/common/git/git.go

```diff
@@ -86,78 +86,49 @@ func FindGitRevision(ctx context.Context, file string) (shortSha string, sha str
 // FindGitRef get the current git ref
 func FindGitRef(ctx context.Context, file string) (string, error) {
 	logger := common.Logger(ctx)
-
 	logger.Debugf("Loading revision from git directory")
-	_, ref, err := FindGitRevision(ctx, file)
-	if err != nil {
-		return "", err
-	}
-
-	logger.Debugf("HEAD points to '%s'", ref)
-
 	// Prefer the git library to iterate over the references and find a matching tag or branch.
-	var refTag = ""
-	var refBranch = ""
+	refName := ""
 	repo, err := git.PlainOpenWithOptions(
 		file,
 		&git.PlainOpenOptions{
 			DetectDotGit:          true,
 			EnableDotGitCommonDir: true,
 		},
 	)
-
 	if err != nil {
 		return "", err
 	}
 
-	iter, err := repo.References()
-	if err != nil {
-		return "", err
-	}
+	headRef, _ := repo.Head()
+	logger.Debugf("HEAD points to revision '%s'", headRef.Hash().String())
 
-	// find the reference that matches the revision's has
-	err = iter.ForEach(func(r *plumbing.Reference) error {
-		/* tags and branches will have the same hash
-		 * when a user checks out a tag, it is not mentioned explicitly
-		 * in the go-git package, we must identify the revision
-		 * then check if any tag matches that revision,
-		 * if so then we checked out a tag
-		 * else we look for branches and if matches,
-		 * it means we checked out a branch
-		 *
-		 * If a branches matches first we must continue and check all tags (all references)
-		 * in case we match with a tag later in the interation
-		 */
-		if r.Hash().String() == ref {
-			if r.Name().IsTag() {
-				refTag = r.Name().String()
-			}
-			if r.Name().IsBranch() {
-				refBranch = r.Name().String()
-			}
-		}
+	// The assumption is made that if the revision has a tag associated with it the ref and ref_name should reference it.
```

Comment:
I’m not confident of when the ref and ref_name are set to the `refs/tags` prefix and therefore whether or not this assumption is accurate or even desired.

I’m thinking this is relevant to working with the [release workflow trigger](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release). 

Either way I’m not looking to change the current behaviour but just fix the value when multiple branches point to the same commit.